client/electron/src/updater.ts

```diff
@@ -448,27 +448,38 @@ export default class AppUpdater {
   }
 
   async checkForUpdates(): Promise<UpdateAvailable | undefined> {
-    if (!this.canCheckForUpdates()) {
-      return;
-    }
-    this.state = UpdaterState.CheckingForUpdate as UpdaterState;
     try {
-      await this.updater.checkForUpdates();
-    } catch (error) {
-      this.lastError = error;
-      this.state = UpdaterState.Idle;
-    }
-    if (this.state === UpdaterState.UpdateAvailable) {
-      return { version: this.lastUpdateInfo!.version };
+      if (!this.canCheckForUpdates()) {
+        return;
+      }
+      this.state = UpdaterState.CheckingForUpdate as UpdaterState;
+      try {
+        await this.updater.checkForUpdates();
+      } catch (error) {
+        this.lastError = error;
+        this.state = UpdaterState.Idle;
+      }
+      if (this.state === UpdaterState.UpdateAvailable) {
+        return { version: this.lastUpdateInfo!.version };
+      }
+    } catch (error: any) {
+      console.error('Could not check for updates', error);
+      return;
     }
   }
 
-  quitAndInstall(): void {
-    if (!this.isUpdateDownloaded()) {
-      console.log('quitAndInstall() called when update has not been downloaded.');
-      return;
+  quitAndInstall(): boolean {
```

Comment:
Why change the return type?

Reply:
`quitAndInstall` can be called following a user action. My initial thought was to be able to warn the user in case it fails, because right now, you could get a situation where the user clicks the button to update and nothing happens. It is quite hard to implement though, so I gave up (for now). At least until we're sure that those try/catch really do something.
I think it can stay as a boolean for later.